packages/esm-patient-immunizations-app/src/immunizations/immunization-utils.ts

```diff
@@ -0,0 +1,17 @@
+import { BehaviorSubject } from 'rxjs';
+
+//This is a work around until attach function is re-worked to enable passing of props
+interface immunizationFormSubProps {
```

Comment:
@KorirC You can ignore this comment for now, it's just a question I personally have and it makes sense to ask it in the context of this PR.

@FlorianRappl This subject here is basically used to transport the current Form props from the overview (which is a separate extension) to the workspace (another extension). Is there a dedicated way to pass data from a component which _opens_ a workspace to _that opened_ workspace? If not, is it okay from your side to use Rx here or would you prefer something else (e.g. a global state)?

Reply:
Definitely **NO** to rxjs.

I'd prefer global state, even though I'd argue a prop passing here would be simpler. The data flow would be

```
component -- (additional args) --> open workspace (using additional args) x--> opened workspace (received args)
(tree 1)                                              (some lib / framework / whatever)                   (tree 2)
```